tests/regression/CMakeLists.txt

```diff
@@ -263,7 +265,8 @@ set_tests_properties("regression/setting_a_buffer_argument_to_NULL_causes_a_segf
   "regression/struct_kernel_arguments" "regression/vector_kernel_arguments"
   "regression/autolocals_in_constexprs" "regression/test_issue_231"
   "regression/test_issue_445" "regression/test_issue_553"
-  "regression/test_issue_577" "regression/test_flatten_barrier_subs"
+  "regression/test_issue_577" "regression/test_issue_757"
```

Comment:
How hard would it be to make the test suite check that this is valgrind clean?

Reply:
I don't know the best way to do this. CTest has a mode in which tests can be run through memcheck. Maybe that could be done automatically, but I'm fluent enough in CMake to figure out how to do that. We'd have to think about how to handle extraneous errors because running the test binary through valgrind still gives a "Conditional jump or move depends on uninitialised value" in LLVM when compiling.